client/tx/tx.go

```diff
@@ -45,21 +45,30 @@ func GenerateOrBroadcastTxWithFactory(clientCtx client.Context, txf Factory, msg
 // simulated and also printed to the same writer before the transaction is
 // printed.
 func GenerateTx(clientCtx client.Context, txf Factory, msgs ...sdk.Msg) error {
+	var updatedTxf Factory
 	if txf.SimulateAndExecute() {
 		if clientCtx.Offline {
 			return errors.New("cannot estimate gas in offline mode")
 		}
 
+		// If we are simulating the transaction, we are not in offline mode. Then we need to lookup correct sequence
+		// number and account number information. Otherwise it will fail
+		txf, err := PrepareFactory(clientCtx, txf)
```

Comment:
@ejfitzgerald thanks for the feedback, I overlooked it and this is why you needed updatedTxf:
The `:=` operator will create variable in current scope and assign to them. Changing to:
```suggestion
		var err error
		txf, err = PrepareFactory(clientCtx, txf)
```
make it reuse the `txf` from the parent scope. But then we have to declare `err` since it doesn't exist yet.

Reply:
Ah yes of course. I just didn't spot that! Nice one